src/_pytest/capture.py

```diff
@@ -136,41 +136,59 @@ def suspend_global_capture(self, in_=False):
 
     def suspend(self, in_=False):
         # Need to undo local capsys-et-al if it exists before disabling global capture.
-        self.suspend_fixture(self._current_item)
+        self.suspend_fixture()
         self.suspend_global_capture(in_)
 
     def resume(self):
         self.resume_global_capture()
-        self.resume_fixture(self._current_item)
+        self.resume_fixture()
 
     def read_global_capture(self):
         return self._global_capturing.readouterr()
 
     # Fixture Control (it's just forwarding, think about removing this later)
 
-    def activate_fixture(self, item):
+    @contextlib.contextmanager
+    def _item_fixture(
```

Comment:
```suggestion
    def _capturing_for_request(
```

I think something like this conveys better what the method is doing.